### PR TITLE
Add ember-source 5 support

### LINKED ISSUE #886: Ember-Source 5
When installing:
```shell
warning " > ember-sinon-qunit@7.1.4" has incorrect peer dependency "ember-source@^3.28.0 || ^4.0.0".
```

Suggested fix for package.json:
```js
"peerDependencies": {
  "ember-source": "^3.28.0 || ^4.0.0 || >=5.0.0",
  "qunit": "^2.0.0",
  "sinon": "^15.0.3 || ^16.0.0"
},
```

### PR DESCRIPTION
**Pull request by hexadecy**
Closes #886

### CODE REVIEW

**Review by elwayman02 — APPROVED** [review]

**Comment by elwayman02** [comment]
I think ember-try is failing and needs to be bumped to 3.0.0 before we can get some clean test builds.

File: package.json [hunk 1]
```diff
@@ -104,23 +104,23 @@
...
   "peerDependencies": {
-    "ember-source": "^3.28.0 || ^4.0.0",
+    "ember-source": "^3.28.0 || ^4.0.0 || >=5.0.0",
     "qunit": "^2.0.0",
-    "sinon": "^15.0.3"
+    "sinon": "^15.0.3 || ^16.0.0"
   },
   "engines": {
-    "node": "14.* || 16.* || >= 18"
+    "node": "16.* || >= 18"
```
Comment (SergeAstapov) [inline package.json:120]: @hexadecy this would be a breaking change - are sure we actually need this as part of this PR?

File: package.json [hunk 2]
```diff
@@ -36,17 +36,18 @@
     "postpack": "ember ts:clean"
   },
   "dependencies": {
-    "@types/sinon": "^10.0.13",
+    "@types/sinon": "^10.0.19",
     "broccoli-funnel": "^3.0.8",
-    "ember-auto-import": "^2.6.1",
-    "ember-cli-babel": "^7.26.11",
+    "ember-auto-import": "^2.6.3",
+    "ember-cli-babel": "^8.2.0",
```
Comment (SergeAstapov) [inline package.json:42]: this is not a req for ember v5, right?

File: package.json [hunk 3]
```diff
@@ -90,12 +90,12 @@
     "ember-data": "^4.11.3",
     "ember-load-initializers": "^2.1.2",
     "ember-page-title": "^7.0.0",
-    "ember-qunit": "^6.2.0",
+    "ember-qunit": "^7.0.0",
```
Comment (elwayman02) [inline package.json:93]: I believe this bump requires us to drop support for Ember 3.28. I'm not opposed to that, necessarily, but I wanted to call out the breaking change here which will require further changes in this PR unless you want to hold off on this bump for now.

File: package.json (hunk 1 above)
Comment (elwayman02) [reply package.json:120]: I'm ok with either approach here. I'm not super gung-ho on keeping Node 14 support when even Node 16 is already EOL at this point as of last month, but if we want to hold off and make that change in a separate release, that's also fine with me.

File: package.json (hunk 3 above)
Comment (SergeAstapov) [reply package.json:93]: > this bump requires us to drop support for Ember 3.28

@elwayman02 not necessarily. we should be able just to use ember-qunit@6 via emebr-try scenario for Ember 3.28 (I did this in couple other addons that still support v3 and it's pretty easy doable)

File: package.json [hunk 4]
```diff
@@ -90,12 +90,12 @@
     "ember-data": "^4.11.3",
```
Comment (elwayman02) [inline package.json:90]: We might need to bump ember-data as well. I'm seeing some failures in the embroider tests due to ember-data, maybe a newer 5.x version works properly?

File: package.json (hunk 1 above)
Comment (SergeAstapov) [reply package.json:120]: @elwayman02 I would say it's just a matter of another major version bump which is extra churn, but indeed not a problem.

as we already planned v2 conversion, I think we can avoid major just to bump supported Node.js version.

I hope I can spend some time today or tomorrow and I'm sure I can make this PR green without requiring new major.

Comment (elwayman02) [reply package.json:120]: Sounds good!

**Comment by elwayman02** [comment]
Thank you for the updates @hexadecy! This is great stuff. Really looking forward to getting this landed.

File: package.json (hunk 2 above)
Comment (hexadecy) [reply package.json:42]: I was having an issue with Babel [babel/preset-env](https://babeljs.io/docs/babel-preset-env) in some tests.

Comment (hexadecy) [reply package.json]: Ok, I rollbacked this one.

**Comment by SergeAstapov** [comment]
@hexadecy @elwayman02 I've reverted breaking changes and opened PR based on this `branch` see #893.

@elwayman02 I believe #893 is good to go as is and can be released as a minor version.

**Pull request closed without merging**